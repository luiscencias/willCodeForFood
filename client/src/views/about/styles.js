import React from 'react';

import {withStyles } from '@material-ui/core/styles';

import Typography from "@material-ui/core/Typography";


export const thumbnailGalleryStyles = {
    background: '#ddd',
    height: '500px',
    width: '1024px',
    margin: '40px auto',
    display: 'flex'
};

export const missionStatement = {
  margin: '40px auto',
  height: '400px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center'
};

export const HeroSection = withStyles(theme => ({
    'section': {
        minHeight: '120px',
        backgroundColor: theme.palette.primary.dark,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center'
    }
}))( ({ children, classes }) => (<section className={classes.section}>{children}</section>) );

export const HeroLogo = withStyles(theme => ({
    root: {
        fontWeight: 400,
        color: theme.palette.secondary.main,
        marginBottom: theme.spacing.unit
    }
}))(Typography);