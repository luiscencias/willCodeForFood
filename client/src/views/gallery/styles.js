import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

export const HeroSection = withStyles(theme => ({
    'section': {
        minHeight: '60vh',
        backgroundColor: theme.palette.primary.dark,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center'
    }
}))( ({ children, classes }) => (<section className={classes.section}>{children}</section>) );

export const HeroLogo = withStyles(theme => ({
    root: {
        fontWeight: 400,
        color: theme.palette.secondary.main,
        marginBottom: theme.spacing.unit
    }
}))(Typography);

export const CallToActionButton = withStyles(theme => ({
    root: {
        borderRadius: '21px',
        width: '120px',

        '&:hover': {
            backgroundColor: theme.palette.secondary.light
        }
    },
    label: {
        color: theme.palette.common.white
    }
}))(Button);

export const SummarySection = withStyles(theme => ({
    'section': {
        display: 'flex',
        paddingTop: theme.spacing.unit * 2,
        paddingRight: theme.spacing.unit * 3,
        paddingLeft: theme.spacing.unit * 3,
        paddingBottom: theme.spacing.unit * 3,

        '& section:nth-child(2)': {
            minWidth: '280px',
            marginLeft: theme.spacing.unit * 3
        }
    }
}))( ({ children, classes }) => (<section className={classes.section}>{children}</section>) );