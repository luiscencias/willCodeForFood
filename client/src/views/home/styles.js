import React from 'react';

import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

export const HeroSection = withStyles(theme => ({
	'section': {
		backgroundColor: theme.palette.primary.dark,
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
		justifyContent: 'center',

		[theme.breakpoints.down('sm')]: {
			height: `calc( 100vh - ${theme.spacing.unit * 7}px )`
		},

		[theme.breakpoints.up('md')]: {
			height: `calc( 100vh - ${theme.spacing.unit * 8}px )`
		}
	}
}))( ({ children, classes }) => (<section className={classes.section}>{children}</section>) );

export const HeroLogo = withStyles(theme => ({
	root: {
		fontWeight: 400,
		color: theme.palette.secondary.main,
		marginBottom: theme.spacing.unit
	}
}))(Typography);

export const CallToActionButton = withStyles(theme => ({
	root: {
		borderRadius: '21px',
		width: '120px',

		'&:hover': {
			backgroundColor: theme.palette.secondary.light
		}
	},
	label: {
		color: theme.palette.common.white
	}
}))(Button);

export const SummarySection = withStyles(theme => ({
	'section': {
		display: 'flex',
		paddingTop: theme.spacing.unit * 2,
		paddingRight: theme.spacing.unit * 3,
		paddingLeft: theme.spacing.unit * 3,
		paddingBottom: theme.spacing.unit * 3,

		'& section:nth-child(2)': {
			minWidth: '280px',
			marginLeft: theme.spacing.unit * 3
		}
	}
}))( ({ children, classes }) => (<section className={classes.section}>{children}</section>) );