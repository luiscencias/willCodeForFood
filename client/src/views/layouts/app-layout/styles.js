import React from 'react';

import { withStyles } from '@material-ui/core/styles';

export const Main = withStyles(theme => ({
	main: {
		minHeight: `calc( 100vh - ${theme.spacing.mainHeader.height}px )`,
		backgroundColor: theme.palette.background.default
	}
}))( ({ children, classes }) => (<main className={classes.main}>{children}</main>) )