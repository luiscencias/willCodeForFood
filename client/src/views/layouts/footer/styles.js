import React from 'react';

import { withStyles } from '@material-ui/core/styles';

export const MainFooter = withStyles(theme => ({
	footer: {
		display: 'flex',
		backgroundColor: theme.palette.primary.light,
		paddingLeft: theme.spacing.unit * 3,
		paddingRight: theme.spacing.unit * 3,
		paddingTop: theme.spacing.unit * 3,
		paddingBottom: theme.spacing.unit * 5
	}
}))( ({ children, classes }) => (<footer className={classes.footer}>{children}</footer>) );