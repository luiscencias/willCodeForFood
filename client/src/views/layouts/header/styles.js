import React from 'react';

import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import Toolbar from '@material-ui/core/Toolbar';

export const spacerStyles = {
	flexGrow: 1
};

export const appBarStyles = {
	boxShadow: 'none'
};

export const MainToolBar = withStyles(theme => ({
	root: {
		height: theme.spacing.mainHeader.height
	}
}))(Toolbar);

export const LogoLink = withStyles(theme => ({
	label: {
		fontWeight: 900
	}
}))(Button);

export const MainNav = withStyles(theme => ({
	'nav': {
		marginRight: theme.spacing.unit * 2
	}
}))( ({ children, classes }) => (<nav className={classes.nav}>{children}</nav>) );

export const MainNavigationLink = withStyles(theme => ({
	root: {
		'&:hover': {
			backgroundColor: 'inherit'
		}
	},

	label: {
		fontWeight: 400,
		textTransform: 'none',

		'&:hover': {
			fontWeight: 500
		}

		/* TODO: media queries */
	}
}))(Button);

export const AuthLink = withStyles(theme => ({
	root: {

		'&$containedSecondary': {
			boxShadow: 'none',
			color: theme.palette.common.white,
			marginLeft: theme.spacing.unit / 2,

			'&:hover': {
				backgroundColor: '#00bcd4'
			}
		}
	},
	label: {
		'&$outlinedSecondary': {

		}
	},
	outlinedSecondary: {

	},
	containedSecondary: {

	}
}))(Button);