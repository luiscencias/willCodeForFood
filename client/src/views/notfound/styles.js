import React from 'react';

import { withStyles } from '@material-ui/core/styles';

import Typography from '@material-ui/core/Typography';

export const HeroSection = withStyles(theme => ({
    'section': {
        minHeight: '80vh',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center'
    }
}))( ({ children, classes }) => (<section className={classes.section}>{children}</section>) );

export const HeroLogo = withStyles(theme => ({
    root: {
        fontWeight: 400,
        //fontSize: 1,
        //display: 'flex',
        //flexDirection: 'row',
        //alignItems: 'center',
        //justifyContent: 'center'
        //color: theme.palette.primary.main,
        //marginBottom: theme.spacing.unit
    }
}))(Typography);
