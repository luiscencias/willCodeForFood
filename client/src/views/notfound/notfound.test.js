import React from 'react';
import { shallow } from 'enzyme';
import {HeroLogo, HeroSection} from './styles'
import NotFound from './notfound'

describe("Not Found", () => {

    let notfound;
    beforeEach(() => notfound = shallow(<NotFound />));

    it('renders', () => {
        expect(notfound.exists()).toBe(true);
    });

    // it('is not authenticated', () => {
    //     console.log(notfound.props);
    //     expect(notfound.props.isAuthenticated).toBeFalsy();
    // });

    it('should\'nt render a div', () => {
        expect(notfound.find('div').length).toEqual(0);
    });

    it('renders without changing', () => {
        expect(notfound).toMatchSnapshot();
        // console.log(notfound.debug());
    });

    it('should have js implementation', () => {
        // console.log(notfound.debug());

        expect(notfound.find('.js')).toBeTruthy();
    });

    // it('should render Hidden Components', () => {
    //     console.log(notfound.debug());
    //
    //     expect(notfound.contains(<Hidden/>));
    // });

    it('should render a HeroSection', () => {
        expect(notfound.contains(<HeroSection/>));

    });

    it('should display HeroLogo', () => {
        // console.log(notfound.debug());

        expect(notfound.contains(<HeroLogo />));
    });

});