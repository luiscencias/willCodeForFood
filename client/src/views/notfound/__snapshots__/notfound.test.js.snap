// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found renders without changing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(WithStyles(Typography))
          variant="h3"
        >
          Page Not Found
        </WithStyles(WithStyles(Typography))>,
        <WithStyles(WithStyles(Typography))
          variant="h6"
        >
          (Wait, how did you get here?)
        </WithStyles(WithStyles(Typography))>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Page Not Found",
          "variant": "h3",
        },
        "ref": null,
        "rendered": "Page Not Found",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "(Wait, how did you get here?)",
          "variant": "h6",
        },
        "ref": null,
        "rendered": "(Wait, how did you get here?)",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(WithStyles(Typography))
            variant="h3"
          >
            Page Not Found
          </WithStyles(WithStyles(Typography))>,
          <WithStyles(WithStyles(Typography))
            variant="h6"
          >
            (Wait, how did you get here?)
          </WithStyles(WithStyles(Typography))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Page Not Found",
            "variant": "h3",
          },
          "ref": null,
          "rendered": "Page Not Found",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "(Wait, how did you get here?)",
            "variant": "h6",
          },
          "ref": null,
          "rendered": "(Wait, how did you get here?)",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
